python/ray/experimental/sgd/interaces.py

```diff
@@ -0,0 +1,134 @@
+from dataclasses import dataclass
+
+from typing import Callable
+
+
+@dataclass
+class WorkerArgs:
+    num_workers: int
+    num_cpus_per_worker: int
+    num_gpus_per_worker: int
+    placement_strategy: str # Assumes 1 bundle per worker.
+
+    elastic_training: bool = False # If True, training will continue with fewer
+    # workers if an actor fails. Also, if more resources are available,
+    # training will continue with more workers.
+    max_actor_restarts: int = 0 # Number of retries when Ray actor fails.
+    # Doesn't do anything if elastic_training is True. Set to -1 for
+    # unlimited restarts.
+    max_failed_actors: int = 0 # If `elastic_training` is True, this
+            # specifies the maximum number of failed actors with which
+            # we still continue training.
```

Comment:
Are you saying create another dataclass for the elastic training args, and then that dataclass is an arg to this? I feel like that would be pretty messy.

Reply:
Yeah, I think typically for people experienced with the API the flattened attributes might be easier (everything is 1 hop away) but for those less familiar nesting helps since it provides logical groupings which are easier to navigate one at a time.